specs/hotel_spec.rb

```diff
@@ -0,0 +1,166 @@
+#require 'pry'
+require_relative 'spec_helper'
+
+
+describe "Hotel Class" do
+  before do
+    @hotel = Hotel::Hotel.new
+  end #before
+
+  describe "Instantiation" do
+    before do
+      @test_hotel = Hotel::Hotel.new
+    end
+
+    it "can create an instance of the Hotel class" do
+      Hotel::Hotel.new.must_be_kind_of Hotel::Hotel
+    end #admin class
+
+    it "can create a hash of rooms" do
+      @test_hotel.all_rooms.must_be_kind_of Hash
+    end #rooms
+
+    it "can return a list of all rooms in the hotel" do
+      @test_hotel.must_respond_to :all_rooms
+      @test_hotel.all_rooms.length.must_equal 20
+    end
+
+  end #Instantiation
+
+  describe "Make Reservations" do
+    it "can call Reservation List to create an array to hold reservations" do
+      @hotel.reservations.must_be_kind_of Hotel::ReservationList
+      @hotel.reservations.reservation_list.must_be_kind_of Array
+    end #array
+
+    it "Can create a Reservation Instance for a specified date range:" do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.reservations.reservation_list[0].must_be_kind_of Hotel::Reservation
+    end #call class
+
+    it "Will add a reservation to the Reservation List:" do
+      @hotel.make_reservation(2222, "2012/12/12", "2012/12/15")
+      @hotel.reservations.reservation_list.length.must_equal 1
+    end #call class
+
+    it "Will Raise an error if asked to reserve an unavailable room" do
+      @hotel.make_reservation(1000, "2001/1/30", "2001/2/4", room: 1)
+      proc {
+        @hotel.make_reservation(1000, "2001/1/30", "2001/2/4", room: 1)
+      }.must_raise ArgumentError
+    end
```

Comment:
This is a good start, but I'd like to see more coverage around what date ranges will conflict with each other. Cases I'd be interested in:
- Not available:
  - Same dates
  - Overlaps in the front
  - Overlaps in the back
  - Completely contained
  - Completely containing
- Available:
  - Completely before
  - Completely after
  - Ends on the checkin date
  - Starts on the checkout date